lib/shoppingcart.rb

```diff
@@ -8,31 +8,28 @@ def initialize(name, capacity)
     @name     = name
     @capacity = capacity.to_i
     @products = []
-
   end
 
   def add_product(item)
     @products << item
   end
 
-
   def details
-    {:name => "King Soopers",
-     :capacity => @capacity}
+    {name: @name, capacity: @capacity}
   end
 
   def total_number_of_products
     @products.sum do |product|
-    product.quantity
+      product.quantity
     end
   end
 
   def is_full?
-    if total_number_of_products > 30
```

Comment:
This is one of the instances where you've hardcoded a value. I also updated it with `>=` so it would still be full if there were exactly 30 items.